### ISSUE TITLE
have a combined ssh+tcp image

### ISSUE THREAD

**Issue by henning-schild** [KSmanis/docker-gentoo-distcc#67]
Currently one has to choose between tcp or ssh but it would likely be possible to have one image that is capable of running both servers at the same time. In there we could maybe even start both services via `openrc` init scripts.

The benefit would be that there would be only one image and users would not have to wonder which one to pick. People could simply not expose the port of the service they do not want to use, or expose both at the same time if they find that useful.

`ssh` has the benefit of encryption, but at the same time it is kind of hard/nasty to configure a gentoo client for that. One has to change the `$HOME` of the portage user and follow several steps to enable that user to `ssh`. To work around the complications of configuring portage for `ssh` i usually use ssh port forwarding and actually use the tcp server via `ssh`. But that trick requires a server to have both ssh and distccd running, which is currently not possible with the container images provided by this project.
Currently i rely on my container host having an ssh server running to apply that trick. But in bigger setups people likely would not be allowed to ssh into the host.

So to sum up there could be several reasons why users might want a combined image. Let me know if that would be interesting and i can propose patches.

**Comment by KSmanis** [comment]
I'm afraid there is no interest in supporting this. As a matter of fact, I have actually been considering discontinuing the `ssh` image variant for some time, because, as you mentioned, configuring portage to use `distccd` over SSH is unnecessarily complicated compared to using TCP connections. As an alternative, I recommend using a VPN (e.g., Tailscale or Point-to-Point WireGuard) or set up SSH port forwarding either on the host or in a sidecar container.

If you feel like experimenting, the `ssh` image variant already bundles `distccd` and `sshd` together, so all you would have to do is override the entrypoint into something that executes `openrc` instead of `sshd`. However I would advise against it, as having a full init system inside a container is way trickier than it sounds; there are multiple projects tackling this issue, going into a lot of trouble to make everything work just right (see https://github.com/just-containers/s6-overlay for instance, as well as the various [LinuxServer](https://www.linuxserver.io/) images that build on top of it).
